### DISCUSSION TITLE
deepin-calculator 的readme.md有误

### DISCUSSION THREAD

**Discussion by xzl01** [linuxdeepin/developer-center#3257] [Q&A | 问答板块] [answered]
我尝试按照readme.md中的指导进行编译，发现两个问题：1.使用的是cmake而不是qmake 2.编译时报错缺少组件，和gtest测试模块有关，需要安装`libgtest-dev` 和 `libgmock-dev`

**Comment by BLumia** [comment, marked as answer]
可以发个 PR 把 README 里的 `qmake` 改成 `cmake`。

关于测试，应该可以在执行 `cmake` 时通过 `-DBUILD_TESTING=OFF` 来关闭测试的构建，可以试试看 :)

**Reply by xzl01** [reply]
我卸载那两个lib之后使用`cmake -DBUILD_TESTING=OFF ..` 进行构建，但是还是报错:
```shell
CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
```

**Reply by BLumia** [reply]
看了一下目前的 [CMakeLists.txt](https://github.com/linuxdeepin/deepin-calculator/blob/b71e7f382660068d6fef2b830263db01bf8bcb5e/CMakeLists.txt#L132) 是无差别包含了 tests 目录，应该判断下 `BUILD_TESTING` 再确定是否包含的，类似这样（ps. 我暂时没测试但应该是这样的）：

``` cmake
if (BUILD_TESTING)
    add_subdirectory(tests)  #tests为ut测试文件夹名
endif ()
```

如果这样好使的话，也可以发个 PR :)

**Reply by xzl01** [reply]
嗯 确信就是这样，我再看看哪里还有问题 一并发了吧，多谢dalao指教
